eos/utils/expression-parser_TEST.cc

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+
+#include <eos/utils/expression.hh>
+#include <eos/utils/expression-fwd.hh>
+#include <eos/utils/expression-maker.cc>
+#include <eos/utils/expression-parser-impl.hh>
+#include <eos/utils/expression-printer.cc>
+#include <eos/utils/kinematic.hh>
+#include <eos/utils/options.hh>
+#include <eos/utils/parameters.hh>
+
+#include <iostream>
+
+using namespace test;
+using namespace eos::exp;
+using namespace eos;
+
+
+class ExpressionEvaluator
+{
+    public:
+        double visit(BinaryExpression & e)
+        {
+            BinaryExpression::func f = BinaryExpression::Method(e._op);
+
+            return f(e._lhs.accept_returning<double>(*this), e._rhs.accept_returning<double>(*this));
+        }
+
+        double visit(ConstantExpression & e)
+        {
+            return e._value;
+        }
+
+        double visit(ObservableExpression & e)
+        {
+            return e._observable->evaluate();
+        }
+
+        double visit(ObservableNameExpression &)
+        {
+            throw InternalError("Encountered ObserableNameExpression in ExpressionEvaluator::visit");
+
+            return 0.0;
+        }
+};
+
+
+struct ExpressionTest
+{
+    const std::string _input;
```

Comment:
The leading underscores indicate that this (and all further members) are private. But they are not. Please remove the leading underscores or make them private.